File: services/account-service/src/main/java/com/banking/platform/accountservice/domain/model/Account.java
package com.banking.platform.accountservice.domain.model;

import lombok.*;
import java.math.BigDecimal;
import java.time.Instant;
import java.util.UUID;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Account {
    private UUID id;
    private UUID customerId;
    private String accountNumber;
    private AccountType type;
    private BigDecimal balance;
    private String currency;
    private AccountStatus status;
    private Instant createdAt;
    private Instant updatedAt;

    public enum AccountType {
        CHECKING, SAVINGS, BUSINESS, JOINT
    }

    public enum AccountStatus {
        ACTIVE, FROZEN, CLOSED, PENDING_ACTIVATION
    }

    public boolean canWithdraw(BigDecimal amount) {
        return this.status == AccountStatus.ACTIVE && this.balance.compareTo(amount) >= 0;
    }

    public boolean isFrozen() {
        return this.status == AccountStatus.FROZEN;
    }
}

File: frontend/src/types/account.ts
import { AccountType, AccountStatus, TransactionType } from './enums';

export interface Account {
  id: string;
  customerId: string;
  accountNumber: string;
  type: AccountType;
  balance: number;
  currency: string;
  status: AccountStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateAccountRequest {
  customerId: string;
  type: AccountType;
  initialBalance: number;
  currency: string;
}

export interface AccountTransaction {
  id: string;
  accountId: string;
  type: TransactionType;
  amount: number;
  currency: string;
  reference?: string;
  balanceAfter: number;
  createdAt: string;
}
